graphsdk/src/androidTest/java/com/microsoft/graph/http/BaseCollectionRequestTests.java

```diff
@@ -0,0 +1,142 @@
+package com.microsoft.graph.http;
+
+import android.test.AndroidTestCase;
+
+import com.microsoft.graph.authentication.MockAuthenticationProvider;
+import com.microsoft.graph.concurrency.MockExecutors;
+import com.microsoft.graph.core.MockBaseClient;
+import com.microsoft.graph.logger.MockLogger;
+import com.microsoft.graph.options.FunctionOption;
+import com.microsoft.graph.options.Option;
+import com.microsoft.graph.options.QueryOption;
+import com.microsoft.graph.serializer.MockSerializer;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Test cases for {@see BaseCollectionRequest}
+ */
+public class BaseCollectionRequestTests extends AndroidTestCase {
+    private MockAuthenticationProvider mAuthenticationProvider;
+    private MockBaseClient mBaseClient;
+
+    @Override
+    protected void setUp() throws Exception {
+        super.setUp();
+        mAuthenticationProvider = new MockAuthenticationProvider();
+        mBaseClient = new MockBaseClient();
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testSend() {
+        final ITestData data = new ITestData() {
+            @Override
+            public int getRequestCode() {
+                return 200;
+            }
+
+            @Override
+            public String getJsonResponse() {
+                return "{ \"id\": \"zzz\" }";
+            }
+
+            @Override
+            public Map<String, String> getHeaders() {
+                final HashMap<String, String> map = new HashMap<>();
+                map.put("Content-Type", "application/octet-stream");
+                return map;
+            }
+        };
+        DefaultHttpProvider mProvider = new DefaultHttpProvider(new MockSerializer(null, ""),
+                mAuthenticationProvider,
+                new MockExecutors(),
+                new MockLogger());
+        mProvider.setConnectionFactory(new MockConnectionFactory(new MockConnection(data)));
+        mBaseClient.setHttpProvider(mProvider);
+        final BaseCollectionRequest request = new BaseCollectionRequest("https://a.b.c", mBaseClient, null, null,null){};
+        request.send();
```

Comment:
Still not validated

---
In reply to: [65750589](https://github.com/microsoftgraph/msgraph-sdk-android/pull/17#discussion_r65750589) [](ancestors = 65750589)